File: youlexuan-manager-web/src/main/java/com/offcn/sellergoods/controller/BrandController.java
package com.offcn.sellergoods.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.offcn.entity.PageResult;
import com.offcn.entity.Result;
import com.offcn.pojo.TbBrand;
import com.offcn.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    BrandService brandService;

    //查询所有，不带有分页
    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }

    //分页查询
    @RequestMapping("/findPage")
    public PageResult findPage(int pageNum , int pageSize){   // url?pageNum=1&pageSize=2
        return brandService.findPage(pageNum,pageSize);
    }

    //@ResponseBody响应成json数据
    //@RequestBody发送过来的json转换成pojo
    @RequestMapping("/add")
    public Result add( @RequestBody TbBrand tbBrand){  //angular传递json过来，将json转换成pojo
        if(tbBrand.getFirstChar().length()>1){
            return new Result(false,"首字母长度不能大于1");
        }
        try {
            brandService.add(tbBrand);
            return new Result(true,"添加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"添加失败");
        }
    }

    @RequestMapping("/findOne")
    public TbBrand findOne( @RequestParam("tbid") Long id){ // ?id
        return brandService.findOne(id);
    }


    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand tbBrand){
        try {
            brandService.update(tbBrand);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改成功");
        }
    }

    @RequestMapping("/delete")
    public Result delete(Long[] ids){
        try {
            brandService.delete(ids);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除成功");
        }
    }


    //带分页的条件查询
    @RequestMapping("/search")
    public PageResult search(int pageNum , int pageSize,@RequestBody TbBrand brand){
        PageResult page = brandService.findPage(pageNum, pageSize, brand);
        return page;
    }

}

File: youlexuan-shop-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController',function ($scope,$http,brandService ,$controller) {


    //声明:brandController需要继承baseControler中的组件
    $controller('baseController',{$scope:$scope});//brandController 继承 baseController中的内容

    $scope.delete = function(){

        //如果数组的长度为0，或者为空
        if($scope.selectIds==null || $scope.selectIds.length==0){
            alert("请选择数据后进行删除");
            return;
        }

        // $http.get("../brand/delete.do?ids="+$scope.selectIds).success(function (resp) {
        brandService.delete($scope.selectIds).success(function (resp) {
            if(resp.success){
                //删除成功，重新加载数据
                $scope.reloadList();
                $scope.selectIds = [];//清空数组
            }else{
                alert(resp.message)
            }
        });
    }


    $scope.findOne = function(id){
        // $http.get("../brand/findOne.do?tbid="+id).success(function (resp) {
        brandService.findOne(id).success(function (resp) {
            //resp
            $scope.entity = resp;
        });
    }


    //添加和修改都进入到save方法
    $scope.save = function(){
        //当entity中数据为空，不允许发送请求

        if($scope.entity.name=='' || $scope.entity.name == null){
            alert("品牌名称不能为空");
            return;
        }
        if($scope.entity.firstChar=='' || $scope.entity.firstChar == null){
            alert("首字母不能为空");
            return;
        }

        /*//根据entity中是否有id，判断是添加还是修改
        var methodName = "add";
        if($scope.entity.id!=null){
            methodName = "update";
        }
        $http.post("../brand/"+methodName+".do",$scope.entity).success(function (resp) {
            //resp
            if(resp.success){
                $scope.entity={name:'',firstChar:''};//清空数据
                //添加成功，重新加载页面数据
                $scope.reloadList();
            }else{
                alert(resp.message);
            }
        });*/

        //根据entity中是否有id，判断是添加还是修改
        if($scope.entity.id!=null){
            // $http.post("../brand/"+methodName+".do",$scope.entity).success(function (resp) {
            brandService.update($scope.entity).success(function (resp) {
                //resp
                if(resp.success){
                    $scope.entity={name:'',firstChar:''};//清空数据
                    //添加成功，重新加载页面数据
                    $scope.reloadList();
                }else{
                    alert(resp.message);
                }
            });
        }else{
            brandService.add($scope.entity).success(function (resp) {
                //resp
                if(resp.success){
                    $scope.entity={name:'',firstChar:''};//清空数据
                    //添加成功，重新加载页面数据
                    $scope.reloadList();
                }else{
                    alert(resp.message);
                }
            });
        }


    }

    //查询所有，不带分页
    $scope.findAll = function () {
        brandService.findAll().success(function (resp) {
            $scope.list = resp;
        });
    };

    //分页查询方法
    //参数1：页码
    //参数2: 每页显示的条数
    $scope.findPage = function(page,rows){
        // $http.get("../brand/findPage.do?pageNum="+page+"&pageSize="+rows).success(function (resp) {
        brandService.findPage(page,rows).success(function (resp) {
            //resp 的结构：   {  total：100，rows:[{},{},{}] }
            $scope.list = resp.rows;
            $scope.paginationConf.totalItems = resp.total;
        });
    };


    //定义查询条件的实体
    $scope.searchEntity = {};
    //带有条件的分页查询
    $scope.search = function(pageNum,pageSize){
        // $http.post("../brand/search.do?pageNum="+pageNum+"&pageSize="+pageSize,$scope.searchEntity).success(function (resp) {
        brandService.search(pageNum,pageSize,$scope.searchEntity).success(function (resp) {
            $scope.paginationConf.totalItems =   resp.total;
            $scope.list =  resp.rows;
        });
    }




});